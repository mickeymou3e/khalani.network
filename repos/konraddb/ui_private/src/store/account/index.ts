export {
  selectApiTokens,
  selectNeutralClient,
  selectNeutralClientCode,
  selectNeutralCustomerCode,
  selectUserIbans,
  selectNeutralUserCode,
  selectIsAdminUser,
} from "./account.selectors";
