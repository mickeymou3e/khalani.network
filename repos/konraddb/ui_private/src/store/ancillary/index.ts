export {
  getPoolStrategies,
  subscribeJasminePoolDeposits,
  subscribeStrategyAssets,
  createStrategy,
  createStrategyAsset,
  selectJasminePoolDeposits,
  selectPoolStrategies,
  selectStrategyAssets,
  ancillaryJasmineApi,
  getAssetBalances,
  transferFunds,
  checkTransactionStatus,
} from "./ancillary.api";

export {
  selectPoolDeposits,
  selectEnergyAttributeTokens,
  seelctSortedEnergyAttributeTokens,
  selectIsEnergyAttributeTokensEmpty,
  selectStrategy,
  selectJltTokenAddress,
  selectSelectedStrategyCode,
  findAttribute,
  selectPoolTokenDepositAddress,
  selectTxFeeDepositAddress,
} from "./ancillary.selectors";

export type {
  EnergyAttributeTokenProps,
  PoolStrategiesResponse,
  PoolStrategy,
  SingleHistoryItem,
  BatchHistoryItem,
  TokenMetadataResponse,
  TokenMetadataAttributes,
  JasminePoolDeposit,
  PoolHistory,
  PoolHistoryResponse,
} from "./ancillary.types";

export { TransferFundsDirection } from "./ancillary.types";

export { Attribute } from "./ancillary.types";

export {
  getHistoryTokenIds,
  addMetadataToHistory,
  getTokenMetadata,
  getBalances,
  topUpTxFeeTokenBalance,
  topUpDefiBalances,
  txFeeMinAmount,
  txFeeMinThreshold,
  checkDefiCallResult,
} from "./ancillary.helpers";

export {
  encodeEnergyAttributeTokenId,
  decodeEnergyAttributeTokenId,
} from "./registry.utils";
