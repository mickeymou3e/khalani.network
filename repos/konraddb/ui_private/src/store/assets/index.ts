export {
  selectSelectedAsset,
  selectFirstAsset,
  selectGuaranteedSelectedAssetAmount,
} from "./assets.selectors";
