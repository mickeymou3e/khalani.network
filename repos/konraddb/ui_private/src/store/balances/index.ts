export {
  selectBalancesList,
  selectPortfolioChartValues,
  selectSelectedAssetAvailableBalance,
  selectCryptoBalancesList,
  selectFiatBalancesList,
  selectIsCryptoBalancesEmpty,
  selectIsFiatBalancesEmpty,
  selectCryptoBalanceAssets,
  selectFiatBalanceAssets,
} from "./balances.selectors";
export type { BalanceData } from "./balances.types";
