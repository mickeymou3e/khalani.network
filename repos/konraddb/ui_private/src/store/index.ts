export {
  persistor,
  setupStore,
  store,
  useAppDispatch,
  useAppSelector,
} from "./store";
export type { AppDispatch, RootState } from "./store";
