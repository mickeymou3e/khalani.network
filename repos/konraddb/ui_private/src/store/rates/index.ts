import {
  selectAllAssetRateDetails,
  selectSelectedAssetRateDetails,
  selectUnauthenticatedAssetRateDetails,
} from "./rates.selectors";
import { RateData } from "./rates.types";

export {
  selectAllAssetRateDetails,
  selectUnauthenticatedAssetRateDetails,
  selectSelectedAssetRateDetails,
};
export type { RateData };
