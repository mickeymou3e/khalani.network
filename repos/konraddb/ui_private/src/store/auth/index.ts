import * as authActions from "./auth.store";

export {
  selectAuthErrorCode,
  selectCsrfToken,
  selectIsTwoFactorAuth,
  selectIsValidLogin,
  selectLoginToken,
  selectWsToken,
  selectBearerToken,
  selectIsNeutralAuthenticated,
  selectIsNotValidLogin,
  selectIsPendingVerification,
  selectNeutralFeatures,
} from "./auth.selectors";
export {
  clearErrorCode,
  clearTokens,
  handleLogout,
  setErrorCode,
  setCsrfToken,
  setLastRequestStartTime,
} from "./auth.store";
export { AuthErrorCodes } from "./auth.types";
export type { AuthState } from "./auth.types";
export { authActions };
