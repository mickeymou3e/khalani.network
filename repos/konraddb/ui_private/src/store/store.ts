import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import {
  combineReducers,
  configureStore,
  PreloadedState,
} from "@reduxjs/toolkit";
import { persistStore } from "redux-persist";

import { bridgeReducer } from "@/features/BridgePage/store/bridge.store";
import { retireReducer } from "@/features/RetirePage/store/retire.store";
import { api } from "@/services/api";
import { graphApi, jasmineApi, neutralApi } from "@/services/extraApis";

import { authReducer } from "./auth/auth.store";
import { backdropsReducer } from "./backdrops/backdrops.store";
import { notificationsReducer } from "./notifications/notifications.store";
import { poolReducer } from "./pool/pool.store";
import { uiReducer } from "./ui/ui.store";
import { walletReducer } from "./wallet/wallet.store";

const rootReducer = combineReducers({
  [api.reducerPath]: api.reducer,
  [jasmineApi.reducerPath]: jasmineApi.reducer,
  [graphApi.reducerPath]: graphApi.reducer,
  [neutralApi.reducerPath]: neutralApi.reducer,
  backdrops: backdropsReducer,
  ui: uiReducer,
  auth: authReducer,
  notifications: notificationsReducer,
  wallet: walletReducer,
  pool: poolReducer,
  retire: retireReducer,
  bridge: bridgeReducer,
});

export const setupStore = (preloadedState?: PreloadedState<RootState>) =>
  configureStore({
    reducer: rootReducer,
    preloadedState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false,
      }).concat(
        api.middleware,
        jasmineApi.middleware,
        graphApi.middleware,
        neutralApi.middleware
      ),
  });

export const store = setupStore();
export const persistor = persistStore(store);

export type RootState = ReturnType<typeof rootReducer>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
