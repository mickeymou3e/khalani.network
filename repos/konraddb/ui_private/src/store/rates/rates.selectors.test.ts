import {
  assignedAssetsMock,
  atomRateMock,
  balancesMock,
  btcRateMock,
  createMockApiInitialState,
  ethRateMock,
  jltRateMock,
} from "@/definitions/__mocks__";
import { AssetName } from "@/definitions/config";
import { selectAssetNameMap } from "@/services/balances";
import {
  selectAllAssetRateDetails,
  selectSelectedAssetRateDetails,
} from "@/store/rates/rates.selectors";
import { setupStore } from "@/store/store";

describe("Rates selectors", () => {
  describe("selectAllAssetRateDetails", () => {
    test("should select all asset rate details", () => {
      const store = setupStore({
        api: createMockApiInitialState({
          rates: {
            BTC: btcRateMock,
            ETH: ethRateMock,
            ATOM: atomRateMock,
            [AssetName["JLT-F23"]]: jltRateMock,
          },
        }),
      });
      const assetNameMap = selectAssetNameMap.resultFunc(balancesMock);
      const allAssetRateDetails = selectAllAssetRateDetails.resultFunc(
        store.getState(),
        assignedAssetsMock[0],
        assetNameMap,
        assignedAssetsMock
      );

      expect(allAssetRateDetails).toMatchSnapshot();
    });
  });

  describe("selectSelectedAssetRateDetails", () => {
    test("should select selected asset rate details", () => {
      const store = setupStore({
        api: createMockApiInitialState({
          rates: {
            BTC: btcRateMock,
            ETH: ethRateMock,
            ATOM: atomRateMock,
          },
        }),
      });
      const assetNameMap = selectAssetNameMap.resultFunc(balancesMock);
      const selectedAssetDetails = selectSelectedAssetRateDetails.resultFunc(
        store.getState(),
        assignedAssetsMock[0],
        assetNameMap
      );

      expect(selectedAssetDetails).toMatchSnapshot();
    });
  });
});
