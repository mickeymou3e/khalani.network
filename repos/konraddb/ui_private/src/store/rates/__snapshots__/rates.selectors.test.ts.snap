// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rates selectors selectAllAssetRateDetails should select all asset rate details 1`] = `
[
  {
    "askPrice": "28,092.65",
    "askPriceUnformatted": "28092.65",
    "askPriceValue": 28092.65,
    "base": "BTC",
    "bidPrice": "27,994.49",
    "bidPriceUnformatted": "27994.49",
    "bidPriceValue": 27994.49,
    "fullName": "Bitcoin",
    "id": "BTC/EUR",
    "marketPrice": "28,043.57",
    "marketPriceValue": 28043.57,
    "pair": "BTC/EUR",
    "quote": "EUR",
    "selected": true,
    "timestamp": 1688593185820,
  },
  {
    "askPrice": "1,761.87",
    "askPriceUnformatted": "1761.87",
    "askPriceValue": 1761.87,
    "base": "ETH",
    "bidPrice": "1,756.58",
    "bidPriceUnformatted": "1756.58",
    "bidPriceValue": 1756.58,
    "fullName": "Ethereum",
    "id": "ETH/EUR",
    "marketPrice": "1,759.23",
    "marketPriceValue": 1759.225,
    "pair": "ETH/EUR",
    "quote": "EUR",
    "selected": false,
    "timestamp": 1688593185916,
  },
  {
    "askPrice": "8.62",
    "askPriceUnformatted": "8.616255025",
    "askPriceValue": 8.616255025,
    "base": "ATOM",
    "bidPrice": "8.59",
    "bidPriceUnformatted": "8.590444975",
    "bidPriceValue": 8.590444975,
    "fullName": "Cosmos",
    "id": "ATOM/EUR",
    "marketPrice": "8.60",
    "marketPriceValue": 8.60335,
    "pair": "ATOM/EUR",
    "quote": "EUR",
    "selected": false,
    "timestamp": 1688593187317,
  },
  {
    "askPrice": "2.04",
    "askPriceUnformatted": "2.04",
    "askPriceValue": 2.04,
    "base": "JLT-F23",
    "bidPrice": "2.02",
    "bidPriceUnformatted": "2.02",
    "bidPriceValue": 2.02,
    "fullName": "JLT-F23",
    "id": "JLT-F23/EUR",
    "marketPrice": "2.03",
    "marketPriceValue": 2.03,
    "pair": "JLT-F23/EUR",
    "quote": "EUR",
    "selected": false,
    "timestamp": 1688593185820,
  },
]
`;

exports[`Rates selectors selectSelectedAssetRateDetails should select selected asset rate details 1`] = `
{
  "askPrice": "28,092.65",
  "askPriceUnformatted": "28092.65",
  "askPriceValue": 28092.65,
  "base": "BTC",
  "bidPrice": "27,994.49",
  "bidPriceUnformatted": "27994.49",
  "bidPriceValue": 27994.49,
  "fullName": "Bitcoin",
  "id": "BTC/EUR",
  "marketPrice": "28,043.57",
  "marketPriceValue": 28043.57,
  "pair": "BTC/EUR",
  "quote": "EUR",
  "timestamp": 1688593185820,
}
`;
